src/main/kotlin/kotlinx/kover/KoverPlugin.kt

```diff
@@ -43,27 +43,22 @@ class KoverPlugin : Plugin<Project> {
     }
 
     private fun Project.createCollectingTask() {
-        tasks.create(COLLECT_TASK_NAME, Copy::class.java) {
+        tasks.create(COLLECT_TASK_NAME, KoverCollectingTask::class.java) {
             it.group = VERIFICATION_GROUP
             it.description = "Collects reports from all submodules in one directory."
-            allprojects { sub ->
-                val xmlReportTask = sub.tasks.withType(KoverXmlReportTask::class.java).getByName(XML_REPORT_TASK_NAME)
+            it.outputDir.set(project.layout.buildDirectory.dir("reports/kover/all"))
+
+            allprojects { p ->
+                val xmlReportTask = p.tasks.withType(KoverXmlReportTask::class.java).getByName(XML_REPORT_TASK_NAME)
                 val htmlReportTask =
-                    sub.tasks.withType(KoverHtmlReportTask::class.java).getByName(HTML_REPORT_TASK_NAME)
+                    p.tasks.withType(KoverHtmlReportTask::class.java).getByName(HTML_REPORT_TASK_NAME)
 
                 it.mustRunAfter(xmlReportTask)
```

Comment:
So it still depends on report tasks and actually invoke tests to collect coverage, right? Then I don't understand what 'remove implicit dependency' means

Reply:
`mustRunAfter` is not dependency, it's ordering rule. It has no effect if the `xmlReportTask` is not scheduled for execution.

Reply:
> Then I don't understand what 'remove implicit dependency' means

The `org.gradle.api.tasks.Copy` task adds an implicit dependency on the task if it uses the output files or directories of that task.

Reply:
OK, I see. So now we copy only files that exist if `report` task was invoked, right?

Reply:
We only copy existing files. The task can be executed, but there may be no tests in it or they are all disabled - in this case, the file does not exist either